xbmc/guilib/GUIFontCache.h

```diff
@@ -39,6 +39,10 @@
 #include "TransformMatrix.h"
 #include "system.h"
 
+#if defined(HAS_GL) || defined(HAS_GLES)
+#include "system_gl.h"
```

Comment:
It's arguably a bigger design issue to take the implementation-independent GUIFontCache.h and introduce GLES and DX specific preprocessing, but that's exactly what was done in the previous commits: https://github.com/xbmc/xbmc/commit/fa6f5d761b600b0a8b52db18168e540b03f79ceb and https://github.com/xbmc/xbmc/commit/e9dd97bac1dc97f9fbdba2b81402aed2c9b92713.

The potential issue here is in assuming the `bufferHandle` is always `unsigned int` instead of correctly using the GL opaque types, and to fix that requires pulling in the right header.  Before these commits, when `bufferHandle` was a `void*` with casting left and right, I spent hours tracking down a GL crash that could have been easily avoided by directly using the GL type as intended.

So yes, I also had qualms about including the header and breaking implementation independence, but that ship sailed two commits ago, and this the safer approach.

What are your suggestions?

Also pinging @Memphiz who's made previous changes to have a look.